bundles/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java

```diff
@@ -159,74 +156,87 @@ private void updateChannels(TimeSeries timeSeries) {
+            int hourOffset = 24 * dayOffset + 12;
+            Optional<Forecast> forecast = timeSeries.getForecast(currentDay, hourOffset);
 
-            if (forecast == null) {
+            if (forecast.isEmpty()) {
                 if (logger.isDebugEnabled()) {
-                    logger.debug("No forecast yet for {}", currentDay.plusHours(offset));
+                    logger.debug("No forecast yet for {}", currentDay.plusHours(hourOffset));
                 }
                 channels.forEach(c -> {
-                    updateState(c.getUID(), UnDefType.NULL);
+                    updateState(c.getUID(), UnDefType.UNDEF);
                 });
             } else {
                 channels.forEach(c -> {
                     String id = c.getUID().getIdWithoutGroup();
-                    BigDecimal value = forecast.getParameter(id);
+                    Optional<BigDecimal> value;
+                    if (isAggregatedChannel(id)) {
+                        value = getAggregatedValue(id, timeSeries, dayOffset);
+                    } else {
+                        value = forecast.get().getParameter(id);
+                    }
                     updateChannel(c, value);
                 });
             }
         }
     }
 
-    private void updateChannel(Channel channel, @Nullable BigDecimal value) {
+    private void updateChannel(Channel channel, Optional<BigDecimal> value) {
         String id = channel.getUID().getIdWithoutGroup();
-        State newState = UnDefType.NULL;
+        State newState = UnDefType.UNDEF;
 
-        if (value != null) {
+        if (value.isPresent()) {
             switch (id) {
                 case PRESSURE:
-                    newState = new QuantityType<>(value, MetricPrefix.HECTO(SIUnits.PASCAL));
+                    newState = new QuantityType<>(value.get(), MetricPrefix.HECTO(SIUnits.PASCAL));
                     break;
                 case TEMPERATURE:
-                    newState = new QuantityType<>(value, SIUnits.CELSIUS);
+                case TEMPERATURE_MAX:
+                case TEMPERATURE_MIN:
+                    newState = new QuantityType<>(value.get(), SIUnits.CELSIUS);
                     break;
                 case VISIBILITY:
-                    newState = new QuantityType<>(value, MetricPrefix.KILO(SIUnits.METRE));
+                    newState = new QuantityType<>(value.get(), MetricPrefix.KILO(SIUnits.METRE));
                     break;
                 case WIND_DIRECTION:
-                    newState = new QuantityType<>(value, Units.DEGREE_ANGLE);
+                    newState = new QuantityType<>(value.get(), Units.DEGREE_ANGLE);
                     break;
                 case WIND_SPEED:
+                case WIND_MAX:
+                case WIND_MIN:
                 case GUST:
-                    newState = new QuantityType<>(value, Units.METRE_PER_SECOND);
+                    newState = new QuantityType<>(value.get(), SIUnits.METRE.divide(Units.SECOND));
```

Comment:
Why did you change this to a custom unit calculation? Same for below.

Reply:
When testing out the binding, I noticed that the values returned from the Item didn't keep the unit that the QuantityType was created with (some with m/s and some with mm/h), but instead used the default, which is km/h for Number:Speed-Items. Since all other types seemed to work correctly (even the Number:Length-Items where some have km an some mm for unit oddly enough) I thought it had something to to with the predefined composite units, so tried to set it explicitly. Forgot to revert afterwards.

Reply:
Reverted now